README.md

```diff
@@ -1,94 +1,120 @@
 ```
 
 
 The general result of `X = FFT(x)` is:
-```text
+```
 X = [(X0r, X0i), (X1r, X1i), (X2r, X2i), (X3r, X3i), (X4r, X4i), (X5r, X5i)]
 ```
 
 However, because our `x` was real-valued, some of this is redundant:
-```text
-FFT(x_c) = [(X0r, 0), (X1r, X1i), (X2r, X2i), (X3r, 0), (X2r, -X2i), (X1r, -X1i)]
+```
+FFT(x) = [(X0r, 0), (X1r, X1i), (X2r, X2i), (X3r, 0), (X2r, -X2i), (X1r, -X1i)]
 ```
 
 As we can see, the output contains a fair bit of redundant data. But it still takes time for the FFT to calculate these values. Converting the input data to complex also takes a little bit of time.
 
-### real-to-complex
+#### real-to-complex
 Using a real-to-complex fft removes the need for converting the input data to complex.
 It also avoids caclulating the redundant output values.
 
-The result is: 
-```text
+The result is:
+```
 RealFFT(x) = [(X0r, 0), (X1r, X1i), (X2r, X2i), (X3r, 0)]
 ```
 
+If the length instead had been 7, result would have been:
+```
+FFT(x) = [(X0r, 0), (X1r, X1i), (X2r, X2i), (X3r, X3i), (X3r, -X3i), (X2r, -X2i), (X1r, -X1i)]
+```
+After removing the reduntant elements, the result is:
+```
+RealFFT(x) = [(X0r, 0), (X1r, X1i), (X2r, X2i), (X3r, X3i)]
+```
+
 This is the data layout output by the real-to-complex fft, and the one expected as input to the complex-to-real ifft.
 
-## Scaling
+### Scaling
 RealFFT matches the behaviour of RustFFT and does not normalize the output of either FFT of iFFT. To get normalized results, each element must be scaled by `1/sqrt(length)`. If the processing involves both an FFT and an iFFT step, it is advisable to merge the two normalization steps to a single, by scaling by `1/length`.
 
-## Documentation
+### Documentation
 
 The full documentation can be generated by rustdoc. To generate and view it run:
-```text
+```
 cargo doc --open
 ```
 
-## Benchmarks
+### Benchmarks
 
 To run a set of benchmarks comparing real-to-complex FFT with standard complex-to-complex, type:
-```text
+```
 cargo bench
 ```
 The results are printed while running, and are compiled into an html report containing much more details.
 To view, open `target/criterion/report/index.html` in a browser.
 
-## Example
+### Example
 Transform a vector, and then inverse transform the result.
 ```rust
-use realfft::{ComplexToReal, RealToComplex};
+use realfft::RealFftPlanner;
 use rustfft::num_complex::Complex;
 use rustfft::num_traits::Zero;
 
-// make dummy input vector, spectrum and output vectors
-let mut indata = vec![0.0f64; 256];
-let mut spectrum: Vec<Complex<f64>> = vec![Complex::zero(); 129];
-let mut outdata: Vec<f64> = vec![0.0; 256];
+let length = 256;
+
+// make a planner
+let mut real_planner = RealFftPlanner::<f64>::new();
 
-//create an FFT and forward transform the input data
-let mut r2c = RealToComplex::<f64>::new(256).unwrap();
+// create an FFT
+let r2c = real_planner.plan_fft_forward(length);
+// make input and output vectors
+let mut indata = r2c.make_input_vec();
```

Comment:
this is a neat idea

Reply:
Yes I think this is quite convenient, so avoid writing the same "let blahblah = vec![.....]" over and over. Something you would like to have in RustFFT as well?